chevah/keycert/ssh.py

```diff
@@ -299,7 +299,7 @@ def fromString(cls, data, type=None, passphrase=None):
             # Most probably some parts are missing from the key, so
             # we consider it too short.
             raise BadKeyError('Key is too short.')
-        except (struct.error,  binascii.Error, TypeError):
+        except (struct.error,  binascii.Error, TypeError, AssertionError):
```

Comment:
all this errors, including the assertion, means that the parsing went wrong. 

in my point of view fromString_SSHCOM_PUBLIC... should not be calling fromString_BLOB the way it is. 
I say this because there is a verification when you call fromString() passing a blob, it calls _guessStringType and there it will only call fromString_BLOB if the blob starts with '\x00\x00\x00\x07ssh-'.. 

so I could check if the blob is valid before trying to extract the type (inside fromString_BLOB) or validate before calling it. i opted for the second option because this way I could keep the error messages consistent when using fromString() passing a OpenSSH or a SSH.COM key. otherwise the error message when trying to parse this kind of keys would be related to the blob, which, in my opinion, could lead to confusion.

still i don't think this code deals in a good way with all the possible errors when parsing keys, but changing it would be a way bigger task then the one I wanted to solve here.  considering that, I think that catching the assertionError here makes sense, as it will lead to raising BadKeyError with a consistent message..